packages/typography/src/InlineCode/InlineCode.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { cx } from '@leafygreen-ui/emotion';
+import {
+  useDarkMode,
+  useUsingKeyboardContext,
+} from '@leafygreen-ui/leafygreen-provider';
+import { codeTypeScaleStyles } from '../styles';
+import { useUpdatedBaseFontSize } from '../utils/useUpdatedBaseFontSize';
+import {
+  anchorClassName,
+  nowrap,
+  normal,
+  code,
+  codeModes,
+  codeLinkStyleModes,
+  codeLinkWrapper,
+  codeFocusModes,
+} from './InlineCode.styles';
+import { InlineCodeProps } from './InlineCode.types';
+
+const InlineCode = React.forwardRef<HTMLElement, InlineCodeProps>(
+  (
+    { children, className, darkMode: darkModeProp, ...rest }: InlineCodeProps,
+    forwardedRef,
+  ) => {
+    const { usingKeyboard: showFocus } = useUsingKeyboardContext();
+    const baseFontSize = useUpdatedBaseFontSize();
+    const { theme } = useDarkMode(darkModeProp);
+    const whiteSpace =
+      ((typeof children === 'string' && children.match(/./gu)?.length) ?? 0) <=
+      30
+        ? nowrap
+        : normal;
```

Comment:
It feels like this should be in the styles file, then referred to from a `cx` call.